File: src/cxx/MsgException.cxx
/*
 * MsgException.cxx
 *
 * Copyright 2006,2007 Martin Quinson, Malek Cherier           
 * All right reserved. 
 *
 * This program is free software; you can redistribute 
 * it and/or modify it under the terms of the license 
 *(GNU LGPL) which comes with this package. 
 *
 */
 
 /* MsgException member functions implementation.
  */  

#include <MsgException.hpp>

#include <string.h>
#include <stdlib.h>
#include <stdio.h>

namespace SimGrid
{
	namespace Msg
	{
		
			MsgException::MsgException()
			{
				this->reason = (char*) calloc(strlen("Internal exception : unknown reason") + 1, sizeof(char));
				strcpy(this->reason, "Internal exception : unknown reason");
			}
		
		
			MsgException::MsgException(const MsgException& rMsgException)
			{
				const char* reason = rMsgException.toString();
				this->reason = (char*) calloc(strlen(reason) + 1, sizeof(char));
				strcpy(this->reason, reason);
				
			}
		
			MsgException::MsgException(const char* reason)
			{
				this->reason = (char*) calloc(strlen("Internal exception : ") + strlen(reason) + 1, sizeof(char));
				sprintf(this->reason, "Invalid exception : %s", reason);
			}
		
		
			MsgException::~MsgException()
			{
				if(this->reason)
					free(this->reason);
			}
				
			const char* MsgException::toString(void) const
			{
				return (const char*)(this->reason);	
			}
		
		
			const MsgException& MsgException::operator = (const MsgException& rMsgException)
			{
				const char* reason = rMsgException.toString();
				this->reason = (char*) calloc(strlen(reason) + 1, sizeof(char));
				
				return *this;
			}
		
	} // namespace Msg	

}// namespace SimGrid

File: src/bindings/java/org/simgrid/msg/MsgException.java
/* This exception is an abstract class grouping all MSG-related exceptions */

/* Copyright (c) 2006-2014. The SimGrid Team.
 * All rights reserved.                                                     */

/* This program is free software; you can redistribute it and/or modify it
 * under the terms of the license (GNU LGPL) which comes with this package. */

package org.simgrid.msg;

/** This exception is an abstract class grouping all MSG-related exceptions */
public abstract class MsgException extends Exception {
	private static final long serialVersionUID = 1L;

	/** Constructs an <code>MsgException</code> without a detail message. */
	public MsgException() {
		super();
	}
	/** Constructs an <code>MsgException</code> with a detail message. */ 
	public MsgException(String msg) {
		super(msg);
	}
}
